cleanlab/experimental/datalab/data.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+"""Module for class and functions that hold and validate datasets that are loaded into DataLab."""
+
+from typing import List, Mapping, Tuple, Union, cast
+
+import datasets
+import numpy as np
+from datasets.arrow_dataset import Dataset
+
+from cleanlab.internal.validation import labels_to_array
+
+
+class Data:
+    """
+    Class that holds and validates datasets for DataLab.
+
+    Parameters
+    ----------
+    data : Dataset
+        Dataset to be used for DataLab.
+
+    label_name : Union[str, List[str]]
+        Name of the label column in the dataset.
+    """
+
+    def __init__(self, data: Dataset, label_name: Union[str, List[str]]) -> None:
```

Comment:
The `datalab.data` module and the associated class is quite new. I just extracted most of the data validation and label extraction functionality to a separate class and used this name as a placeholder.

I want to hold off on changing the variable name because I don't necessarily think we will stick to the `: Dataset` annotation. I hope we can use some data-like interface that's more generic.

Reply:
Makes sense